### DISCUSSION TITLE
How to pass default user and password (postgres) to docker start

### DISCUSSION THREAD

**Discussion by KIC** [GreptimeTeam/greptimedb#1298] [Q&A] [answered]
I followed the documentation but I can not connect via the postgres protocol because I don't know the default credentials. The documentations seems not to cover how I can set user/password for the docker installation: https://docs.greptime.com/getting-started/installation/greptimedb-standalone#docker

Any hints are appreciated, thank you very much!

**Comment by shuiyisong** [comment, marked as answer]
Hi
By default greptimedb does't setup any password. However, if you want to connect via PostgreSQL protocol, you'll need to specify the database you want to connect to, for example `psql -h localhost -p 4003 -d public`

You can refer to this doc for more detail: https://docs.greptime.com/user-guide/protocols/postgresql#connect

**Reply by KIC** [reply]
Thanks.

Interesting, I started the container with:
```
docker run -p 4003:4003 \
-v "$(pwd)/greptimedb:/tmp/greptimedb" \
greptime/greptimedb standalone start \
```

And when I connect to database "greptimedb" it it cant connect. But I can get an authentication failed error if I provide some user. It seems that like for some reason datagrip wants to use a user and password. Can I pass the user password settings to the container like you could with:

```
cargo run -- standalone start --user-provider=static_user_provider:cmd:greptime_user=greptime_pwd
```

**Reply by KIC** [reply]
commandline also seems to fail:
![image](https://user-images.githubusercontent.com/10957396/229109920-318e3928-6ab7-4621-b540-ab510d6b553f.png)

```
sudo netstat -tulpn | grep 4003
tcp        0      0 0.0.0.0:4003            0.0.0.0:*               LISTEN      18066/docker-proxy  
tcp6       0      0 :::4003                 :::*                    LISTEN      18073/docker-proxy
```

**Reply by shuiyisong** [reply]
Hi
According to this [doc](https://docs.greptime.com/user-guide/protocols/postgresql#connect), you'll need to specify `--postgres-addr 0.0.0.0:4003` too. The full command would be 

```
docker run -p 4003:4003 \
      -v "$(pwd)/greptimedb:/tmp/greptimedb" \
      --name greptime --rm \
      greptime/greptimedb standalone start \
      --postgres-addr 0.0.0.0:4003
```

Once the server is started, you should be able to connect without providing any username or password. Here's my command line screenshot

<img width="824" alt="image" src="https://user-images.githubusercontent.com/113876041/229269314-00f31f9f-0ce1-4923-a504-8a3e60927c6b.png">
<img width="824" alt="image" src="https://user-images.githubusercontent.com/113876041/229269318-bb06439e-686f-43bd-8e76-20e9fd25a26c.png">

I tried to connect using datagrip but failed. My guess is that datagrip pre-executing some sql to get database information that might be not yet supported by greptimedb. Here's my test connection screenshot

<img width="912" alt="image" src="https://user-images.githubusercontent.com/113876041/229269768-a9f855e3-b6c1-4d49-b248-7eafe77b87e6.png">

**Reply by KIC** [reply]
indeed I totally missed the: --postgres-addr 0.0.0.0:4003

thanks!
